tests/integration/test_dynamic_sampling.py

```diff
@@ -589,3 +599,198 @@ def test_relay_chain(
 
     envelope = mini_sentry.captured_events.get(timeout=1)
     envelope.get_transaction_event()
+
+
+@mock.patch.dict(
+    GLOBAL_CONFIG,
+    {
+        "options": {
+            "profiling.profile_metrics.unsampled_profiles.platforms": ["python"],
+            "profiling.profile_metrics.unsampled_profiles.sample_rate": 1.0,
+            "profiling.profile_metrics.unsampled_profiles.enabled": True,
+        }
+    },
+)
+def test_relay_chain_keep_unsampled_profile(
+    mini_sentry,
+    relay,
+):
+
+    # Create an envelope with a profile:
+    def make_envelope(public_key):
+        trace_uuid = "414e119d37694a32869f9d81b76a0bbb"
+        transaction_uuid = "414e119d37694a32869f9d81b76a0baa"
+
+        envelope, trace_id, event_id = _create_transaction_envelope(
+            public_key,
+            trace_id=trace_uuid,
+            event_id=transaction_uuid,
+            platform="python",
+        )
+        te = envelope.get_transaction_event()
+        profile_payload = get_profile_payload(te)
+        envelope.add_item(
+            Item(
+                payload=PayloadRef(bytes=json.dumps(profile_payload).encode()),
+                type="profile",
+            )
+        )
+        return envelope
+
+    project_id = 42
+    relay = relay(relay(mini_sentry))
+    config = mini_sentry.add_basic_project_config(project_id)
+    config["config"]["transactionMetrics"] = {"version": 1}
+    config["config"]["features"] = ["projects:profiling-ingest-unsampled-profiles"]
+
+    public_key = config["publicKeys"][0]["publicKey"]
+    SAMPLE_RATE = 0.0
+    _add_sampling_config(config, sample_rate=SAMPLE_RATE, rule_type="transaction")
+
+    envelope = make_envelope(public_key)
+
+    relay.send_envelope(project_id, envelope)
+    envelope = mini_sentry.captured_events.get(timeout=1)
+    envelope = mini_sentry.captured_events.get(timeout=1)
+
+    profiles = list(
+        filter(lambda item: item.data_category == "profile", envelope.items)
+    )
+    assert len(profiles) == 1
+    assert not profiles[0].headers.get("sampled")
```

Comment:
We're looking for a very specific value, so I would make this check more strict:

```suggestion
    assert profiles[0].headers["sampled"] is False
```